### PR TITLE
Fix ortho

### PR DESCRIPTION
**Pull request by pablocael**
This should be merged after https://github.com/rdiankov/openrave/pull/854 since its branched from it

### CODE REVIEW

**Comment by rdiankov** [comment]
@braineo did you review this?

**Comment by pablocael** [comment]
@rdiankov , this is still now reviewd. This branch depends on another one called manipImprov which should be merged first. That manipImprov was mostly reviewed, but Im still implementing some last things on it. After finishing manipImprov, this one can be reviewed.

**Comment by braineo** [comment]
@rdiankov No I have not checked this branch yet. Track mode fix should come first then this one.

**Comment by rdiankov** [comment]
@pablocael can you resolve the conflicts? thanks

File: plugins/qtosgrave/osgviewerwidget.cpp [hunk 1]
```diff
@@ -1452,36 +1500,46 @@ void QOSGViewerWidget::SetCurrentCameraManipulator(osgGA::CameraManipulator* man
...
 
-void QOSGViewerWidget::SetCurrentManipulatorDistanceToFocus(double distance)
+void QOSGViewerWidget::SetCameraDistanceToFocus(double distance)
 {
     if( distance <= 0 ) {
         return;
     }
+
+    if(IsInOrthoMode()) {
+        RAVELOG_WARN("performing hard SetCameraDistanceToFocus when in ortho mode, please use Zoom() to change zoom independently from the current view mode.");
```
Comment (braineo) [inline plugins/qtosgrave/osgviewerwidget.cpp:1516]: Should have early return after warning message

File: plugins/qtosgrave/qtosgviewer.cpp [hunk 2]
```diff
@@ -148,8 +148,8 @@ QtOSGViewer::QtOSGViewer(EnvironmentBasePtr penv, std::istream& sinput) : QMainW
                     "Accepts 0/1 value that decides whether to render the cross hairs");
     RegisterCommand("SetNearPlane", boost::bind(&QtOSGViewer::_SetNearPlaneCommand, this, _1, _2),
                     "Sets the near plane for rendering of the image. Useful when tweaking rendering units");
-    RegisterCommand("SetTextureCubeMap", boost::bind(&QtOSGViewer::_SetTextureCubeMap, this, _1, _2),
-                    "Sets the skybox with cubemap");
+    // RegisterCommand("SetTextureCubeMap", boost::bind(&QtOSGViewer::_SetTextureCubeMap, this, _1, _2),
```
Comment (braineo) [inline plugins/qtosgrave/qtosgviewer.cpp:151]: Why it is commented out?

File: plugins/qtosgrave/qtosgviewer.cpp [hunk 3]
```diff
@@ -1426,7 +1427,7 @@ bool QtOSGViewer::_TrackLink(KinBody::LinkPtr link, const RaveTransform<float>&
     if (infoText.size() == 0) {
         infoText = str(boost::format("%s/%s")%parentIem->GetName()%link->GetName());
     }
-    _posgWidget->TrackNode(osgNode, infoText, linkOffset, _posgWidget->GetCurrentManipulatorDistanceToFocus());
+    _posgWidget->TrackNode(osgNode.get(), infoText, linkOffset, _posgWidget->GetCameraDistanceToFocus());
```
Comment (braineo) [inline plugins/qtosgrave/qtosgviewer.cpp:1430]: Do you need `osgNode.get()`?

**Review by braineo — CHANGES_REQUESTED** [review]

File: plugins/qtosgrave/osgviewerwidget.cpp [hunk 4]
```diff
@@ -1155,7 +1197,8 @@ void QOSGViewerWidget::_PanCameraTowardsDirection(double delta, const osg::Vec3d
 
 void QOSGViewerWidget::MoveCameraZoom(float factor, bool isPan, float panDelta)
 {
-    if(isPan) {
+    std::cout << "MOVE CAMERA ZOOM factor = " << factor << " isPan = " << isPan << " panDelta = "  << panDelta << "is ortho = " << IsInOrthoMode() << std::endl;
```
Comment (braineo) [inline plugins/qtosgrave/osgviewerwidget.cpp]: Use `RAVELOG`, probably verbose

**Review by braineo — CHANGES_REQUESTED** [review]

File: plugins/qtosgrave/qtosgviewer.cpp (hunk 2 above)
Comment (pablocael) [reply plugins/qtosgrave/qtosgviewer.cpp]: Reenabled :)

File: plugins/qtosgrave/osgviewerwidget.cpp (hunk 4 above)
Comment (pablocael) [reply plugins/qtosgrave/osgviewerwidget.cpp]: I actually removed those message now, they were debug and I forgot to remove.

File: plugins/qtosgrave/osgviewerwidget.cpp (hunk 1 above)
Comment (pablocael) [reply plugins/qtosgrave/osgviewerwidget.cpp:1514]: So maybe an error is better? a warning suggest that its still ok to do but not adisable... I think..

**Comment by pablocael** [comment]
@braineo Finished the camera view point command.. Now I will connect it to the ui.

File: plugins/qtosgrave/osgviewerwidget.h [hunk 5]
```diff
@@ -320,6 +325,7 @@ class QOSGViewerWidget : public QOpenGLWidget
     double _zNear; ///< In OSG, znear and zfar are updated by CullVisitor, which
                    ///  causing getProjectionMatrixAsXXX to return negative
                    ///  values. Therefore, we manage zNear ourselves
+    double _currentOrthoFrustumSize;
```
Comment (braineo) [inline plugins/qtosgrave/osgviewerwidget.h:339]: Documentation for this member variable

File: plugins/qtosgrave/osgviewerwidget.cpp (hunk 1 above)
Comment (braineo) [reply plugins/qtosgrave/osgviewerwidget.cpp:1583]: I think warn is also ok

**Review by braineo — APPROVED** [review]

**Comment by rdiankov** [comment]
thanks~

**Pull request merged**